bundles/EcommerceFrameworkBundle/IndexService/Worker/ElasticSearch/AbstractElasticSearch.php

```diff
@@ -836,4 +840,126 @@ protected function createEsIndex(string $indexName)
+        return $this->getIndexVersion()+1;
+    }
+
+    /**
+     * Build the name of the index for a specific index number.
+     * @param int $id e.g. 13
+     * @return string the name of the index, such as at_de_elastic_13
+     */
+    private function buildIndexName(int $id) : string {
+        $currentIndexVersion = $this->getIndexVersion();
+        $this->indexVersion = $id;
+        $indexName = $this->getIndexNameVersion();
+        $this->indexVersion = $currentIndexVersion;
+        return $indexName;
+    }
+
+    /**
+     * Create an index with the given name (delete existing ones), and force a rebuild in ES.
+     * Does not perform an index switch.
+     * @param string $indexName
+     * @throws BadRequest400Exception
+     * @throws \Elasticsearch\Common\Exceptions\NoNodesAvailableException
+     */
+    private function buildIndexAndPerformEsRebuild(string $indexName) {
+        $indexName = strtolower($indexName);
+        $esClient = $this->getElasticSearchClient();
+        Logger::info('Index '.$indexName.' already existed. Delete and rebuild it...');
+        $result = $esClient->indices()->exists(['index' => $indexName]);
+        if ($result) {
+            Logger::info('Deleted '.$indexName.'.');
+            $esClient->indices()->delete(['index' => $indexName]);
+        }
+
+        $this->createEsIndex($indexName);
+
+        $body =
+            [
+                'source' => [
+                    'index' => $this->getIndexNameVersion(), //currently active ES index with data.
+                    /*
+                    "query" => [
+                        "term" => [
+                            "attributes.remoteId" => "12J928143"
+                        ]
+                    ] */
+                ],
+                'dest' => [
+                    // 'index' => 'mytestindex',
+                    'index' => $indexName, //target index name
+                ]
+            ];
+
+        try {
+
+
+            $startTime = time();
+            Logger::info('Start reindexing process in Elastic Search...', [
+                'method' => 'POST',
+                'uri' => '/_reindex',
+                'body' => $body
+            ]);
+            $result = $esClient->transport->performRequest('POST', '/_reindex', [], $body);
+            Logger::info(sprintf('Completed re-index in %.02f seconds.', (time() - $startTime)));
+
+        } catch (BadRequest400Exception $e) {
+            //could not reindex.
+            throw $e;
+        }
+    }
+
+    /**
+     * Perform a native reindexing via the ES reindexing API.
+     * This is typically much faster than reindexing single records.
+     * @throws \Exception
+     */
+    public function performNativeReindexing() {
+        $hasReindexIndexingModeCompleted = false;
+        Lock::lock(self::REINDEXING_LOCK_KEY);
+        if ($this->isInReindexMode()) {
+            //stop reindexing process. If not yet finished, no alias switch will take place
```

Comment:
this does not stop the reindex-mode .. it just checks, if it is finished, and if so, it switches alias.

Reply:
so whats the point of these lines?